### PR TITLE
implemented the leaked cards

### PR DESCRIPTION
**Pull request by Sarvesh025**
image

### CODE REVIEW

File: src/app/leaked_credentials/page.tsx [hunk 1]
```diff
@@ -1,45 +1,46 @@
...
     const [wsSavedEmails, setWsSavedEmails] = useState<string[]>([]);
     const [leakedEmails, setLeakedEmails] = useState();
+    const [leakedEmailDetails, setLeakedEmailDetails] = useState<any>([]);
     const [email, setEmail] = useState({
         addEmail: '',
         searchEmail: ''
     });
     const [emailActive, setEmailActive] = useState('');
-    const [loader, setLoader] = useState(true);
+    const [loader, setLoader] = useState(false);
```
Comment (arijeet-mukherjee) [inline src/app/leaked_credentials/page.tsx:25]: set loader true initially

File: src/app/leaked_credentials/page.tsx [hunk 2]
```diff
@@ -83,7 +84,18 @@ export default function leaked_credentials() {
     }
 
     function handleEmailActive(email: any) {
-        setEmailActive(email);
+
+        try {
+            makeWebServiceCall(`${API_BASE_URL}/api/leakCredentialDetails/getAllLeakedCredDetailsByCondition?email=${email}&is_fixed=no`, 'GET', "", jwtToken)
```
Comment (arijeet-mukherjee) [inline src/app/leaked_credentials/page.tsx:89]: why an api call here ? It is not required

**Review by arijeet-mukherjee — CHANGES_REQUESTED** [review]
Dont make unnecessary api calls. api call mostly done at useEffect level or some event handlers. Take inspiration from leak cred page how I did to refactor your code

File: src/app/leaked_credentials/page.tsx (hunk 2 above)
Comment (Sarvesh025) [reply src/app/leaked_credentials/page.tsx:89]: changed the function name

File: src/app/leaked_credentials/page.tsx (hunk 1 above)
Comment (Sarvesh025) [reply src/app/leaked_credentials/page.tsx]: resolved in latest commit

File: src/component/LeakedDetailCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                   
+                }
+                catch (err) {
+                    console.error(err);
+                    alert('Error occurred while updating Leaked email state');
+                }
+            })();
+        }
+        else{
+            (async function(){
```
[the remark this answers is not in the document]
Comment (Sarvesh025) [reply src/component/LeakedDetailCard/index.tsx]: removed IIFE and implemented disable function on check box

**Review by arijeet-mukherjee — APPROVED** [review]
LGTM

**Pull request merged**